cmd/podman/system_df.go

```diff
@@ -0,0 +1,363 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/containers/libpod/cmd/podman/cliconfig"
+	"github.com/containers/libpod/cmd/podman/formats"
+	"github.com/containers/libpod/cmd/podman/libpodruntime"
+	"github.com/containers/libpod/libpod"
+	"github.com/containers/libpod/libpod/image"
+	units "github.com/docker/go-units"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+var (
+	dfSystemCommand     cliconfig.SystemDfValues
+	dfSystemDescription = `
+	podman system df
+
+	Show podman disk usage
+	`
+	_dfSystemCommand = &cobra.Command{
+		Use:   "df",
+		Short: "Show podman disk usage",
+		Long:  dfSystemDescription,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			dfSystemCommand.GlobalFlags = MainGlobalOpts
+			return dfSystemCmd(&dfSystemCommand)
+		},
+	}
+)
+
+func init() {
+	dfSystemCommand.Command = _dfSystemCommand
+	dfSystemCommand.SetUsageTemplate(UsageTemplate())
+}
+
+var systemDfHeader = map[string]string{
+	"Type":        "TYPE",
+	"TotalCount":  "TOTAL",
+	"Active":      "ACTIVE",
+	"Size":        "SIZE",
+	"Reclaimable": "RECLAIMABLE",
+}
+
+type systemDfDiskUsage struct {
+	Type        string
+	TotalCount  int
+	Active      int
+	Size        string
+	Reclaimable string
+}
+
+func dfSystemCmd(c *cliconfig.SystemDfValues) error {
+	runtime, err := libpodruntime.GetRuntime(&c.PodmanCommand)
+	if err != nil {
+		return errors.Wrapf(err, "Could not get runtime")
+	}
+	defer runtime.Shutdown(false)
+
+	ctx := getContext()
+
+	systemDfDiskUsages, err := getDiskUsage(ctx, runtime)
+	if err != nil {
+		return errors.Wrapf(err, "error getting output of system df")
+	}
+	out := formats.StdoutTemplateArray{Output: systemDfDiskUsageToGeneric(systemDfDiskUsages), Template: "table {{.Type}}\t{{.TotalCount}}\t{{.Active}}\t{{.Size}}\t{{.Reclaimable}}", Fields: systemDfHeader}
```

Comment:
@baude Was this what we threw away in `ps` because of perf concerns?

Reply:
ping @baude